### ISSUE TITLE
Qwen3-4B e2e greedy baseline mismatch on main

### ISSUE THREAD

**Issue by xiaguan** [xiaguan/pegainfer#188]
## Summary

`cargo test --release -p pegainfer-qwen3-4b --test e2e` fails on current main (de26afa) — greedy output diverges from `test_data/Qwen3-4B.json` baseline.

## Reproduction

```bash
PEGAINFER_TEST_MODEL_PATH=/data/models/Qwen3-4B cargo test --release -p pegainfer-qwen3-4b --test e2e
```

## Observed

"Tell me a story" case:
- **Got**: `" about a young girl named Lila who is a talented painter, but she is not allowed to paint because of her family's rules..."`
- **Expected** (baseline): `" about a young girl who is a talented artist, but she is not allowed to paint because of her gender..."`

Diverges from the first generated token after "a young girl".

## Notes

- Baseline was last refreshed at 6b55d64 (2026-05-25)
- `fix(qwen3): align bf16 greedy numerics` (5f72a90, 2026-05-26) landed after the baseline refresh — likely the source
- Either re-verify numerics and update the baseline, or revert the regression

**Comment by scatyf3** [comment]
Hi, can you assign this to me?

**Comment by xiaguan** [comment]
Sure, thanks for the contribution.

**Comment by xiaguan** [comment]
By the way, this precision issue is quite subtle and may well fall within acceptable tolerances. However, if you're interested in exploring it further, I guess it might be related to some FlashInfer operators.

**Comment by xiaguan** [comment]
Closing as superseded: the exact-text e2e baseline this issue is about was deleted in #194 and replaced by the HF logits golden gate (`tests/hf_golden_gate.rs`), which compares logprob deltas against a stored HF bf16 golden instead of exact strings — the mismatch mode described here can no longer occur. Qwen3.5 still has the exact-text gate; its replacement is tracked in #186 / #249.
